server/api/availability_handler.py

```diff
@@ -10,10 +10,12 @@
 
 availability_handler = Blueprint('availability_handler', __name__)
 
-@availability_handler.route("/availability/<int:id>", methods=["GET"])
+
+@availability_handler.route("/availability/<url>", methods=["GET"])
 @check_token
-def get_calendar_availability(id):
-    user = Users.query.filter_by(id=id).first()
+def get_calendar_availability(url):
+    uid = EventTypes.query.filter_by(url=url).first().user_id
+    user = Users.query.filter_by(id=uid).first()
```

Comment:
To save some typing, can combine two lines into user = EventTypes.query.filter_by(url=url).first().user